encodings/dict/src/compute/mod.rs

```diff
@@ -89,14 +89,59 @@ mod test {
 
     #[test]
     fn canonicalise_nullable_primitive() {
-        let reference =
-            PrimitiveArray::from_option_iter([Some(42), Some(-9), None, Some(42), None, Some(-9)]);
-        let dict = dict_encode(reference.as_ref()).unwrap();
-        let flattened_dict = dict.into_array().into_primitive().unwrap();
-        // Compressor puts 0 as a code for invalid values which we end up using in take
-        // thus invalid values on decompression turn into whatever is at 0th position in dictionary
-        assert_eq!(flattened_dict.as_slice::<i32>(), [42, -9, 42, 42, 42, -9]);
-        assert_eq!(flattened_dict.validity_mask().unwrap().true_count(), 4);
+        let values: Vec<Option<i32>> = (0..65)
+            .map(|i| match i % 3 {
+                0 => Some(42),
+                1 => Some(-9),
+                2 => None,
+                _ => unreachable!(),
+            })
+            .collect();
+
+        let dict = dict_encode(PrimitiveArray::from_option_iter(values.clone()).as_ref()).unwrap();
+        let actual = dict.into_array().into_primitive().unwrap();
+
+        let expected: Vec<i32> = (0..65)
+            .map(|i| match i % 3 {
+                // Compressor puts 0 as a code for invalid values which we end up using in take
```

Comment:
You should expect any value between the min and max value in that array

Reply:
This test is testing nullability. The crucial thing to test here is 
```
                0 | 2 => 42,
                1 => -9,
```

Reply:
This test isn't hit by the SIMD implementation yet. Might as well consider removing it.